src/algebra/pointwise.lean

```diff
@@ -511,6 +560,57 @@ lemma mem_smul_of_mem [has_scalar α β] {t : set β} {a} {b} (ha : a ∈ s) (hb
   a • b ∈ s • t :=
 ⟨a, b, ha, hb, rfl⟩
 
+@[simp, to_additive]
+lemma smul_empty [has_scalar α β] (s : set α) : s • (∅ : set β) = ∅ :=
+by rw [←image2_smul, image2_empty_right]
+
+@[simp, to_additive]
+lemma empty_smul [has_scalar α β] (t : set β) : (∅ : set α) • t = ∅ :=
+by rw [←image2_smul, image2_empty_left]
+
+@[simp, to_additive]
+lemma inter_smul_subset [has_scalar α β] {s₁ s₂ : set α} {t : set β} :
+  (s₁ ∩ s₂) • t ⊆ s₁ • t ∩ s₂ • t :=
+λ x ⟨a, b, ha, hb, hx⟩, ⟨⟨a, b, ha.1, hb, hx⟩, ⟨a, b, ha.2, hb, hx⟩⟩
+
+@[simp, to_additive]
+lemma smul_inter_subset [has_scalar α β] {s : set α} {t₁ t₂ : set β} :
+  s • (t₁ ∩ t₂) ⊆ s • t₁ ∩ s • t₂ :=
+λ x ⟨a, b, ha, hb, hx⟩, ⟨⟨a, b, ha, hb.1, hx⟩, ⟨a, b, ha, hb.2, hx⟩⟩
+
+@[simp, to_additive]
+lemma union_smul [has_scalar α β] {s₁ s₂ : set α} {t : set β} : (s₁ ∪ s₂) • t = s₁ • t ∪ s₂ • t :=
+by simp_rw [←image2_smul, image2_union_left]
+
+@[simp, to_additive]
+lemma smul_union [has_scalar α β] {s : set α} {t₁ t₂ : set β} : s • (t₁ ∪ t₂) = s • t₁ ∪ s • t₂ :=
+by simp_rw [←image2_smul, image2_union_right]
+
+@[to_additive]
+lemma Union_smul {ι : Sort*} [has_scalar α β] (f : ι → set α) (t : set β) :
```

Comment:
Can we have Inter versions of these?

Reply:
They are false.

Reply:
Yeah, we have `inter_smul_subset`, might as well have the non-binary version!